src/workerd/jsg/ser.h

```diff
@@ -60,9 +146,16 @@ class Serializer final: v8::ValueSerializer::Delegate {
 // the stack.
 class Deserializer final: v8::ValueDeserializer::Delegate {
 public:
+  // Exactly like Serializer::ExternalHandler, but for Deserializer.
+  class ExternalHandler {
+  public:
+    virtual ~ExternalHandler() noexcept(false) = 0;
+  };
+
   struct Options {
     kj::Maybe<uint32_t> version;
     bool readHeader = true;
+    kj::Maybe<ExternalHandler&> externalHandler;
```

Comment:
nit: a comment here identifying the owner of the `ExternalHandler` instance would be helpful